jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/impl/admin/UserTaskAdminServiceImplTest.java

```diff
@@ -161,6 +165,72 @@ public void testAddPotentialOwnersNotBusinessAdmin() {
+        processInstanceId = processService.startProcess(deploymentUnit.getIdentifier(), "org.jbpm.writedocument");
+        assertNotNull(processInstanceId);
+
+        List<TaskSummary> tasks = runtimeDataService.getTasksAssignedAsPotentialOwner("salaboy", new QueryFilter());
+        assertEquals(1, tasks.size());
+        TaskSummary task = tasks.get(0);
+
+        userTaskService.release(task.getId(), "salaboy");
+        userTaskAdminService.addPotentialOwners(15456, false, factory.newUser("john"));
+    }
+
+    @Test
+    public void testAddRemovePotentialOwnersAsGroup() {
+        processInstanceId = processService.startProcess(deploymentUnit.getIdentifier(), "org.jbpm.writedocument");
+        assertNotNull(processInstanceId);
+
+        List<TaskSummary> tasks = runtimeDataService.getTasksAssignedAsPotentialOwner("salaboy", new QueryFilter());
+        assertEquals(1, tasks.size());
+        TaskSummary task = tasks.get(0);
+        userTaskService.release(task.getId(), "salaboy");
+
+        // Forward the task to HR group (Add HR as potential owners)
+        identityProvider.setRoles(Arrays.asList("HR"));
+        userTaskAdminService.addPotentialOwners(task.getId(), true, factory.newGroup("HR"));
+        tasks = runtimeDataService.getTasksAssignedAsPotentialOwner("katy", new QueryFilter());
+        assertEquals(1, tasks.size());
+
+        // HR has no resources to handle so lets forward it to accounting
+        userTaskAdminService.removePotentialOwners(task.getId(), factory.newGroup("HR"));
+        tasks = runtimeDataService.getTasksAssignedAsPotentialOwner("katy", new QueryFilter());
+        assertEquals(0, tasks.size());
+
+        identityProvider.setRoles(Arrays.asList("Accounting"));
+        userTaskAdminService.addPotentialOwners(task.getId(), false, factory.newGroup("Accounting"));
+        tasks = runtimeDataService.getTasksAssignedAsPotentialOwner("mary", new QueryFilter());
+        assertEquals(1, tasks.size());
+    }
+
+    @Test
+    public void testAddRemoveExcludedOwnersAsGroup() {
+        processInstanceId = processService.startProcess(deploymentUnit.getIdentifier(), "org.jbpm.writedocument");
+        assertNotNull(processInstanceId);
+
+        List<TaskSummary> tasks = runtimeDataService.getTasksAssignedAsPotentialOwner("salaboy", new QueryFilter());
+        assertEquals(1, tasks.size());
+        TaskSummary task = tasks.get(0);
+        userTaskService.release(task.getId(), "salaboy");
+
+        identityProvider.setRoles(Arrays.asList("IT"));
+        userTaskAdminService.addExcludedOwners(task.getId(),true, factory.newGroup("IT"));
+        tasks = runtimeDataService.getTasksAssignedAsPotentialOwner("salaboy", new QueryFilter());
+        assertEquals(0, tasks.size());
+
+        userTaskAdminService.removeExcludedOwners(task.getId(), factory.newGroup("IT"));
+        tasks = runtimeDataService.getTasksAssignedAsPotentialOwner("salaboy", new QueryFilter());
+        assertEquals(1, tasks.size());
```

Comment:
`AssertJ` will make this much nicer, including the failure message - `Assertions.assertThat(tasks).hasSize(1);`
This could be done on many places in this PR.